### ISSUE TITLE
v1.4.0-alpha.2: no network connectivity with network bonding

### ISSUE THREAD

**Issue by uhthomas** [siderolabs/talos#6973]
### Description

There is no network connectivity with [v1.4.0-alpha.2](https://github.com/siderolabs/talos/releases/tag/v1.4.0-alpha.2) as talos is not able to make the bond.

### Logs

```
[talos] controller failed: {"component": "controller-runtime", "controller": "network.LinkSpecController", "error": "2 errors occurred:\n\t* error enslaving/unslaving link \"eth4\" under \"bond0\": netlink receive: operation not permitted\n\t* error enslaving/unslaving link \"eth5\" under \"bond0\": netlink receive: operation not permitted\n\n"}
```

### Environment

- Talos version: v1.4.0-alpha.2
- Kubernetes version: v1.26.2
- Platform: bar metal (Dell R430)

**Comment by smira** [comment]
Probably it makes sense to post network config

**Comment by uhthomas** [comment]
@smira Sure, sorry. I didn't feel it was necessary given it works fine with v1.3.6.

```json
 {
    "op": "add",
    "path": "/machine/network",
    "value": {
        "interfaces": [
            {
                "interface": "bond0",
                "addresses": [
                    "10.0.0.101"
                ],
                "bond": {
                    "downdelay": 200,
                    "lacpRate": "fast",
                    "miimon": 100,
                    "mode": "802.3ad",
                    "updelay": 200,
                    "interfaces": [
                        "eth4",
                        "eth5"
                    ],
                    "xmitHashPolicy": "layer3+4"
                },
                "dhcp": true,
                "vip": {
                    "ip": "10.0.0.100"
                }
            }
        ]
    }
}
```

**Comment by smira** [comment]
I wonder if your network link names got changed or something to that effect. I don't see any other immediate issue I can think of

**Comment by uhthomas** [comment]
Possibly? Though they do seem consistent across reboots with older versions. The kernel did change from 5 to 6, so may have an effect?

This would of course go away with https://github.com/siderolabs/talos/issues/6756.

**Comment by strygin** [comment]
Same problem here after the upgrade from stable with the same mc. Seems like `controller-runtime` needs CAP_NET_ADMIN capability:
https://github.com/torvalds/linux/blob/b2bc47e9b2011a183f9d3d3454a294a938082fb9/net/core/rtnetlink.c#L2720
Could you please add capabilities `setcap` to the config?

**Comment by frezbo** [comment]
controller-runtime already has those capabilities, this is nothing related to capabilities.

**Comment by smira** [comment]
bug with Linux 6.1, fixed in #7028
